test/evergreen.yml

```diff
@@ -2883,9 +2881,9 @@ tasks:
     exec_timeout_secs: 25200
     commands:
       - func: "get project"
-      - func: "compile wiredtiger"
+      - func: "compile wiredtiger address sanitizer"
         vars: 
-          posix_configure_flags: -DENABLE_STRICT=1 -DHAVE_DIAGNOSTIC=1 -DNON_BARRIER_DIAGNOSTIC_YIELDS=1 -DHAVE_BUILTIN_EXTENSION_LZ4=1 -DHAVE_BUILTIN_EXTENSION_SNAPPY=1 -DHAVE_BUILTIN_EXTENSION_ZLIB=1 -DHAVE_BUILTIN_EXTENSION_ZSTD=1
```

Comment:
`compile wiredtiger address sanitizer` uses `-DHAVE_BUILTIN_EXTENSION_ZLIB=1` while here we have `-DHAVE_BUILTIN_EXTENSION_ZSTD=1`, is that an issue? Same question for `race-condition-stress-sanitizer-test`.

Reply:
I believe ZSTD is faster than ZLIB but does same / similar thing. I will add in DHAVE_BUILTIN_EXTENSION_ZSTD=1 as it seems like it was being used before. Thanks for spotting!